.github/workflows/main.yaml

```diff
@@ -78,45 +78,45 @@ jobs:
         uses: docker/build-push-action@v5
         with:
           context: .
-
-# temporarily comment out to get GH action into place.  Will fix tests on a follow on PR.
-# https://github.com/Sceptre/sceptre/pull/1412#discussion_r1452566731
-#  integration-tests:
-#    needs:
-#      - unit-tests
-#    if: ${{ github.event_name != 'pull_request' }}
-#    runs-on: ubuntu-latest
-#    permissions:
-#      id-token: write
-#    steps:
-#      - uses: actions/checkout@v4
-#      - name: Install Poetry
-#        uses: snok/install-poetry@v1
-#      - name: Install dependencies
-#        run: poetry install --no-interaction --all-extras
-#      # Update poetry for https://github.com/python-poetry/poetry/issues/7184
-#      - name: update poetry
-#        run: poetry self update --no-ansi
-#      - name: Setup Python
-#        id: setup-python
-#        uses: actions/setup-python@v5
-#        with:
-#          python-version: '3.10'
-#          cache: 'poetry'
-#      - name: Assume AWS role
-#        uses: aws-actions/configure-aws-credentials@v4
-#        with:
-#          aws-region: ${{ env.AWS_REGION }}
-#          role-to-assume: ${{ env.AWS_ROLE }}
-#          role-session-name: GHA-${{ github.repository_owner }}-${{ github.event.repository.name }}-${{ github.run_id }}
-#          role-duration-seconds: ${{ env.AWS_ROLE_DURATION }}
-#      - name: run tests
-#        run: poetry run behave integration-tests/features --junit --junit-directory build/behave
-
-  docker-build-push:
+  integration-tests:
     needs:
+      - linting
+      - packaging
+      - documentation
+      - unit-tests
       - docker-build
-    if: ${{ github.ref == 'refs/heads/master' }}
+    if: ${{ github.ref_type == 'push' }}
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+      - name: Install dependencies
+        run: poetry install --no-interaction --all-extras
+      # Update poetry for https://github.com/python-poetry/poetry/issues/7184
+      - name: update poetry
+        run: poetry self update --no-ansi
+      - name: Setup Python
+        id: setup-python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+          cache: 'poetry'
+      - name: Assume AWS role
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: ${{ env.AWS_REGION }}
+          role-to-assume: ${{ env.AWS_ROLE }}
+          role-session-name: GHA-${{ github.repository_owner }}-${{ github.event.repository.name }}-${{ github.run_id }}
+          role-duration-seconds: ${{ env.AWS_ROLE_DURATION }}
+      - name: run tests
+        run: AWS_DEFAULT_REGION=eu-west-1 poetry run behave integration-tests/features --junit --junit-directory build/behave
```

Comment:
Another (perhaps supplemental) option is to set the region for all steps, with an `env:` command: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#env

Reply:
I don't think that's possible due to way we've setup the AWS test account.  The roles and OIDC provider access is setup in us-east-1 so when i attempt to manually login using eu-west-1 region i get an error  message that says `Invalid_grant provided`.  I'm guessing the same thing will happen when assuming a role with `aws-actions/configure-aws-credentials` using eu-west-1.   That's why we login with us-east-1 then execute tests in eu-west-1

Reply:
If we can't set it globally, then it would certainly be more idiomatic to do it this way:

```yaml
  - name: run tests
    run: poetry run behave integration-tests/features --junit --junit-directory build/behave
Member
    env:
      AWS_DEFAULT_REGION: eu-west-1
```

Reply:
done